contracts/XVSVault/XVSVault.sol

```diff
@@ -755,12 +775,15 @@ contract XVSVault is XVSVaultStorage, ECDSA, AccessControlledV5 {
     }
 
     function _writeCheckpoint(address delegatee, uint32 nCheckpoints, uint96 oldVotes, uint96 newVotes) internal {
-        uint32 blockNumber = safe32(block.number, "XVSVault::_writeCheckpoint: block number exceeds 32 bits");
+        uint32 blockNumberOrSecond = safe32(
```

Comment:
Shouldn't we be more careful when we are using block.timestamp, as uint32 may fail to handle it in the future?

Reply:
Users may still want to be able to withdraw their money after 70 years though